src/main/scala/util/gen.scala

```diff
@@ -190,4 +190,106 @@ object gen {
           OpenCL.Compilation.SeparateHostAndKernelCode.separate(hostFunName),
           (OpenCL.Module.apply _).tupled)
   }
+
+  object cuda {
```

Comment:
cuda code generation duplicates a lot of code. For this i also created: `shine.cuda.KernelModule` and `shine.cuda.Module`

Reply:
We will look at this in the context of #106